### PR TITLE
Enable hive text reads by default

### PR DESCRIPTION
**Pull request by revans2**
Now that we have fixed all of the big issues with hive text file reads we should enable this by default. We have configs to be able to disable specific data types that we know could have issues. This also updates the docs to explain those differences.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala [hunk 1]
```diff
@@ -246,6 +246,34 @@ class HiveProviderImpl extends HiveProvider {
...
+              willNotWorkOnGpu("reading of doubles has been disabled set " +
+                  s"${RapidsConf.ENABLE_READ_HIVE_DOUBLES} to true to enable this.")
+            }
+
+            lazy val hasDecimals = wrapped.output.exists { att =>
+              TrampolineUtil.dataTypeExistsRecursively(att.dataType,
+                dt => dt.isInstanceOf[DecimalType])
+            }
+
+            if (!conf.shouldHiveReadDecimals && hasDecimals) {
```
Comment (mythrocks) [inline sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala:273]: My gut reaction to this was to consider a switch for `TIMESTAMP`. The only known issue is in the case of `timestamp.formats`, which is its own switch.

**Comment by revans2** [comment]
The test that failed is to verify that hive is off by default. I forgot to update the test.  Working on it...

File: sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala (hunk 1 above)
Comment (revans2) [reply sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala:273]: I will add it.

**Comment by mythrocks** [comment]
The test fails because of a bug introduced in #7265.
I have raised #7303 for this. A PR is up in #7305.

**Review by mythrocks — APPROVED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@mythrocks please take another look I think I have fixed all of the review comments.

**Review by mythrocks — APPROVED** [review]
Sorry, I thought I'd approved this already.

**Pull request merged**